src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -202,7 +211,18 @@ private bool IsOpenProperty(ODataProperty property)
 
             WriteInstanceAnnotation(property, isTopLevel, currentPropertyInfo.MetadataType.IsUndeclaredProperty);
 
-            ODataValue value = property.ODataValue;
+            WriterValidationUtils.ValidatePropertyDefined(this.currentPropertyInfo, this.MessageWriterSettings.ThrowOnUndeclaredPropertyForNonOpenType);
+
+            duplicatePropertyNameChecker.ValidatePropertyUniqueness(property);
+

```

Comment:
Not sure why these were moved? Can we put the two checks back before line 206 (before we write the type annotation for undeclared properties)?

Reply:
Moved back to before L206. Thanks.

---
In reply to: [233183156](https://github.com/OData/odata.net/pull/1106#discussion_r233183156) [](ancestors = 233183156)